bin/save_cache

```diff
@@ -48,7 +48,17 @@ if ! aws s3api head-object --bucket "$S3_BUCKET_NAME" --key "$CACHE_KEY" > /dev/
 	echo "No existing cache entry for $CACHE_KEY – storing in cache"
 
 	echo "	Compressing"
-	tar -czf "$CACHE_KEY" "$CACHE_FILE"
+	TAR_CONFIG=${4-}
+	if [[ "$TAR_CONFIG" == '--use_alternate_tar_config' ]]; then
```

Comment:
Nit but I'm not a fan of the name of this custom flag… what about `--use_relative_path_in_tar` or something like that instead?

---

Also, I'm wondering if it wouldn't make sense to adopt that `-C` behavior… in all circumstances after all (seems a nicer solution than including `/User/builder` in the cache path and risking the un-tar-ing to fail if used on a different builder which might not have a `/User/builder` folder (all our Mac VMs currently do, but we shouldn't need to make that assumption, and also nothing says the cache would be restored on a Mac VM vs an Android AMI or Docker container or whatnot, so…).

That one might be a broader investigation to verify that changing that behavior that `save_cache` was using so far (and you kept in the `else`) to instead use `-C "$CACHE_FILE" .` in all cases would not break existing setups depending on how they use `restore_cache` on their end; so could be saved for a subsequent PR if you're not sure about that. But would be a nice simplification if we can confirm it and go in that direction.

Reply:
Agreed, I like `--use_relative_path_in_tar` better, too. Changed in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/43/commits/b919255f6778f69af21855091369fc4caa96dad6

Reply:
> I'm wondering if it wouldn't make sense to adopt that -C behavior in all circumstances. 

I agree that would be nice to use the same behavior for all cases if possible. Because I saw different behavior between the existing `tar` configuration versus the one I added in this PR, I had assumed the projects that use `save_cache` would need to be updated and cause a breaking change. I haven't tested that out yet, but I think it'd be outside the scope for this project. I'll create a separate issue for it.